Assets/BossRoom/Scripts/Shared/Net/ConnectionManagement/ServerGameNetPortal.cs

```diff
@@ -174,6 +174,15 @@ private void ApprovalCheck(byte[] connectionData, ulong clientId, NetworkManager
             string payload = System.Text.Encoding.UTF8.GetString(connectionData);
             var connectionPayload = JsonUtility.FromJson<ConnectionPayload>(payload); // https://docs.unity3d.com/2020.2/Documentation/Manual/JSONSerialization.html
 
+            if (connectionPayload.isDebug != Debug.isDebugBuild)
+            {
+                gameReturnStatus = ConnectStatus.IncompatibleBuildType;

```

Comment:
I'm curious if we have to still invoke the connectionApprovalCallback in the cases where we're denying the client's connection. This is me just thinking that perhaps these are the cases where the client can't host afterwards? If there's an explicit reason why, I suppose I'd like to know why (and users would too).

Reply:
I asked around and it turns out the WaitToDisconnect coroutine is a workaround we had to do to be able to send a disconnect reason message to the client, as if they don't receive messages after they are disconnected/their connection is denied. A ticket has been created for that on the sdk side.

So I will use the same workaround, but add comments to make it clearer.

As for the bug preventing clients to host afterwards, I tested this scenario to be sure and did not encounter issues. The bug is really a UI issue, caused by the same thing that PR #460 is looking to solve